weaviate/collections/batch/base.py

```diff
@@ -161,27 +155,41 @@ def __init__(
         self.__results_for_wrapper = results
         self.__results_lock = threading.Lock()
 
-        self.__retry_failed_objects: bool = False
-        self.__retry_failed_references: bool = False
-
         self.__cluster = Cluster(self.__connection)
 
+        self.__batching_mode: _BatchMode = batch_mode
         self.__max_batch_size: int = 1000
-        self.__dynamic_batching = fixed_batch_size is None
-        self.__recommended_num_objects: int = (
-            fixed_batch_size if fixed_batch_size is not None else 10
-        )
+
+        if isinstance(self.__batching_mode, _FixedSizeBatching):
+            self.__recommended_num_objects = self.__batching_mode.batch_size
+            self.__concurrent_requests = self.__batching_mode.concurrent_requests
+        elif isinstance(self.__batching_mode, _RateLimitedBatching):
+            self.__concurrent_requests = (
+                self.__batching_mode.requests_per_minute + self.__max_batch_size
+            ) // self.__max_batch_size
```

Comment:
From physicist to physicist, the units here don't match but is this necessarily a problem though? (The first term is a rate while the second a scalar)

Reply:
oh yeah, this is just a fancy way of doing ìnt(ceil(self.__batching_mode.requests_per_minute / self.__max_batch_size))`, I can change if you prefer :)

I reused some parameters to not have 50 different ones that are only used by some batching modes, so the names are not always great :/ 
I added a bit more explanation, not sure if the physics approach helps with understanding.

Reply:
Ah, okay I didn't know of this shortcut. It's good to know now! My concern was only a gut reaction from years of scolding by lecturers :)